matcha/onnx/infer.py

```diff
@@ -0,0 +1,168 @@
+import argparse
+import os
+import warnings
+from pathlib import Path
+from time import perf_counter
+
+import numpy as np
+import onnxruntime as ort
+import soundfile as sf
+import torch
+
+from matcha.cli import plot_spectrogram_to_numpy, process_text
+
+
+def validate_args(args):
+    assert (
+        args.text or args.file
+    ), "Either text or file must be provided Matcha-T(ea)TTS need sometext to whisk the waveforms."
+    assert args.temperature >= 0, "Sampling temperature cannot be negative"
+    assert args.speaking_rate >= 0, "Speaking rate must be greater than 0"
+    return args
+
+
+def write_wavs(model, inputs, output_dir, vocoder=None):
+    if vocoder is None:
+        print("The provided model has the vocoder embedded in the graph.\nGenerating waveform directly")
```

Comment:
A bit confused here, if the vocoder is None, how are we getting waveforms directly here?

Reply:
I though it was returning mel, and mel_lengths as in the export.py.

`model, output_names = matcha, ["mel", "mel_lengths"]`

Reply:
It is a naming issue, the correct name is `external_vocoder`. If it is `None`, this means that the vocoder is already embedded in the ONNX graph.